File: dsl-output/ArticleService.java
package model;
public interface ArticleService{
  public model.Article retrieveById(java.util.UUID id) throws java.lang.Throwable;
  public java.util.List<model.Article> retrieveAll() throws java.lang.Throwable;
  public void add(model.Article article) throws java.lang.Throwable;
  public void save(java.util.UUID id, model.Article article) throws java.lang.Throwable;
  public void remove(java.util.UUID id) throws java.lang.Throwable;
}

File: dsl-output/ArticleService.js
angular.module("app", ['ngRoute']).service("ArticleService", ['$http', ArticleService]);
function ArticleService($http){
  this.retrieveById = function(id){
    return $http.get('/api/articles/'+id+'')
  }
  this.retrieveAll = function(){
    return $http.get('/api/articles')
  }
  this.add = function(article){
    return $http.post('/api/articles', article)
  }
  this.save = function(id, article){
    return $http.put('/api/articles/'+id+'', article)
  }
  this.remove = function(id){
    return $http.delete('/api/articles/'+id+'')
  }
}
